common/beerocks/bcl/source/beerocks_ucc_listener.cpp

```diff
@@ -790,30 +691,27 @@ void beerocks_ucc_listener::handle_wfa_ca_command(const std::string &command)
             break;
         }
 
-        // Check if CMDU message type has preprepared CMDU which can be loaded
+        // Check if CMDU message type has preprepared CMDU which can be loaded -
+        // If so, create CMDU manually by loading CMDU from preperared buffer
         auto certification_tx_buffer = get_buffer_filled_with_cmdu();
-        if (!certification_tx_buffer) {
-            LOG(ERROR) << "certification_tx_buffer is not allocated";
-            reply_ucc(eWfaCaStatus::ERROR, err_internal);
-            break;
-        }
-        ieee1905_1::CmduMessageTx external_cmdu_tx(
-            certification_tx_buffer.get() + sizeof(beerocks::message::sUdsHeader),
-            message::MESSAGE_BUFFER_LENGTH - sizeof(beerocks::message::sUdsHeader));
-        if (create_cmdu(external_cmdu_tx, ieee1905_1::eMessageType(message_type), err_string)) {
+        if (certification_tx_buffer) {
```

Comment:
Ok, now I understand why you have decided to remove the buffer after each use. Still, I think a better idea (instead of deallocating the buffer) will be to add a flag like "read_ready" that will be set to true after the buffer is filled, and that being cleared after the buffer is read by 
`get_buffer_filled_with_cmdu()`. If calling to `get_buffer_filled_with_cmdu()` when the "read_ready" is cleared, than return a nullptr.

Reply:
You are again paying more attention to performance than to code readability - in this case there is absolutly no reason to do that. The code is simpler to understand when certification buffer is allocated and used only for one message then freed. Performance is not an issue here, especially that this happens only in certification.